johnzon-mapper/src/main/java/org/apache/johnzon/mapper/Mappings.java

```diff
@@ -98,8 +119,47 @@ public Boolean isDeduplicateObjects() {
             return deduplicateObjects;
         }
 
+        @Override
+        public MappingType getType() {
+            return MappingType.CLASSMAPPING;
+        }
+
     }
 
+    public static class ParameterizedTypeMapping extends TypeMapping {
```

Comment:
this class should be able to provides its parameters as TypeMapping to not require reflection at runtime no?

Reply:
what ? Isn't this what I am actually doing ?

Reply:
Not really cause it means the instances of done at https://github.com/apache/johnzon/pull/20/files#diff-7f654235e6be2d22d4eb2df20cd40b19R220 could be replaced by a findTypeMapping directly (then a switch on the type and the new TypeMapping hierarchy should allow to align all the implementations with our old CollectionMapping (https://github.com/amoscatelli/johnzon/blob/bfbed85a045bcb53c071e77d01a1b52ca29d6c70/johnzon-mapper/src/main/java/org/apache/johnzon/mapper/MappingParserImpl.java#L187) instead of creating a hierarchy we actually don't use.